dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/programrule/DefaultProgramRuleService.java

```diff
@@ -202,10 +212,22 @@ private List<TrackedEntityAttributeValue> getAttributes(
   // if they are present in both places
   private Set<Event> getEventsFromEnrollment(
       String enrollmentUid, TrackerBundle bundle, TrackerPreheat preheat) {
-    Stream<Event> events =
-        preheat.getEvents().values().stream()
-            .filter(e -> e.getEnrollment().getUid().equals(enrollmentUid))
-            .filter(e -> bundle.findEventByUid(e.getUid()).isEmpty());
+    Stream<Event> events;
+    try {
+      events =
+          eventService
+              .getEvents(
+                  EventOperationParams.builder()
+                      .eventParams(EventParams.TRUE)
+                      .orgUnitMode(ACCESSIBLE)
+                      .enrollments(Set.of(enrollmentUid))
+                      .build())
+              .stream()
+              .filter(e -> e.getEnrollment().getUid().equals(enrollmentUid))
```

Comment:
Why is this necessary? 🤔 should we not only get events for this enrollment as you put that into the params?

Reply:
It is not, copy and paste code :grimacing: